src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -32,26 +34,106 @@ void check_level_zero_attributes_match(const IONodeDescriptor& nodeDescriptor,
     }
 
-    for (size_t index = 0; index < ovDimensions.size(); ++index) {
-        if (ovDimensions[index] != zeDescriptor.info.dims[index] && !nodeDescriptor.originalShape.is_dynamic()) {
+    for (size_t index = ovDimensions.size(); index < ZE_MAX_GRAPH_ARGUMENT_DIMENSIONS_SIZE; ++index) {
+        if (zeDescriptor.info.dims[index] != 0 && zeDescriptor.info.dims[index] != 1) {
             OPENVINO_THROW("Shape mismatch for parameter " + name);
         }
     }
-    for (size_t index = ovDimensions.size(); index < ZE_MAX_GRAPH_ARGUMENT_DIMENSIONS_SIZE; ++index) {
-        if (zeDescriptor.info.dims[index] != 0 && zeDescriptor.info.dims[index] != 1) {
+
+    for (size_t index = 0; index < ovDimensions.size(); ++index) {
+        if (ovDimensions[index] != zeDescriptor.info.dims[index] && !nodeDescriptor.transposedShape.is_dynamic()) {
             OPENVINO_THROW("Shape mismatch for parameter " + name);
         }
     }
 }
 
+std::optional<size_t> get_batch_size_for_node(const IONodeDescriptor& nodeDescriptor,
+                                              const ZeroExecutor::ArgumentDescriptor& zeDescriptor) {
+    Logger logger("GetBatchSizeForNode", Logger::global().level());
+
+    const std::vector<size_t>& ovDimensions = nodeDescriptor.originalShape.get_shape();
+    switch (zeDescriptor.info.deviceLayout) {
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NCHW:
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NHWC:
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NCDHW:
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NDHWC:
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NC:
+        if ((ovDimensions[BATCH_AXIS] == zeDescriptor.info.dims[BATCH_AXIS]) &&
+            (ovDimensions[BATCH_AXIS] != DEFAULT_BATCH_SIZE)) {
+            logger.info("Batching on the plugin is not used, batching is handled by the compiler");
+            return std::nullopt;
+        } else {
+            return ovDimensions[BATCH_AXIS];
+        }
+        break;
+    default:
+        logger.info("Batching on the plugin is working only when batching is found on 0th dimension");
+        return std::nullopt;
+    }
+
+    return DEFAULT_BATCH_SIZE;
+}
+
+/**
+ * @brief Get the batch size to be handled on the plugin.
+ * @details Analyze the shape from the compiled model with the shape from the originalShape and get the originalShape if
+ * it is different.
+ * @param metadata A map to represent descriptions for inputs and outputs of a network.
+ * @param executorInputDescriptors A map to represent Level zero inputs descriptors.
+ * @param executorOutputDescriptors A map to represent Level zero outputs descriptors.
+ */
+
+std::optional<size_t> get_batch_size(
+    const NetworkMetadata& metadata,
+    const std::unordered_map<std::string, ZeroExecutor::ArgumentDescriptor>& executorInputDescriptors,
+    const std::unordered_map<std::string, ZeroExecutor::ArgumentDescriptor>& executorOutputDescriptors) {
+    std::set<size_t> batch_size;
+
+    Logger logger("GetBatchSize", Logger::global().level());
```

Comment:
Nitpick:

Can you consider renaming `snake_case` to `camelCase` for the functions `get_batch_size` and `get_batch_size_for_node`. You already use `camelCase` in `logger`.

Reply:
done, thanks!